openeo_processes_dask/process_implementations/cubes/merge.py

```diff
@@ -120,21 +106,12 @@ def merge_cubes(
             c2 = cube1
         check = []
         for c in c1.dims:
-            check.append(c in c1.dims and c in c2.dims)
-        for c in c2.dims:
-            if not (c in c1.dims):
-                dimension = c
-        if (
-            np.array(check).all()
-            and len(c2[dimension]) == 1
-            and callable(overlap_resolver)
-        ):
-            c2 = c2.transpose(dimension, ...)
-            merge = overlap_resolver(c2[0], c1, **context)
-        elif isinstance(overlap_resolver, xr.core.dataarray.DataArray):
-            merge = overlap_resolver
+            check.append(c in c2.dims) # dims of smaller cube have to be in larger one
+        if all(check) and callable(overlap_resolver):
+            parameters = {"x": cube1, "y": cube2, "context": context}
+            merge = overlap_resolver(parameters=parameters)
         else:
-            raise Exception("OverlapResolverMissing")
+            raise Exception('OverlapResolverMissing')
```

Comment:
This wants to be a new exception type in `openeo_processes_dask:exceptions.py` :)